src/foolscap/connections/tor.py

```diff
@@ -0,0 +1,167 @@
+import os, re
+from twisted.internet.interfaces import IStreamClientEndpoint
+from twisted.internet.defer import inlineCallbacks, returnValue, succeed
+import ipaddress
+from .. import observer
+
+from zope.interface import implementer
+from ..ipb import IConnectionHintHandler, InvalidHintError
+from ..util import allocate_tcp_port
+import txtorcon
```

Comment:
Yeah, it's in Foolscap mostly for testing purposes, but 'flogtool web' needs to open a listening port. The foolscap version is synchronous (just stdlib `socket` calls), which we don't need here, but it also has some extra complexity to avoid some funny OS behavior that would otherwise return a port that's really in use.